### PR TITLE
Draft Unix help

### PR DESCRIPTION
**Pull request by visionjo**
Work in progress

Still a work in progress (I need to go over and finalize in the morning before assigning reviewers)

### CODE REVIEW

File: docs/source/using-discovery/bash.rst [hunk 1]
```diff
@@ -0,0 +1,553 @@
+.. _using_bash:
+
+***********
+Using Bash
+***********
+Overview
+=================
+Bash (Bourne Again SHell) is a popular shell and command-line interface in Unix-like operating systems. A shell is an interface between the user and the underlying operating system, allowing users to interact with the system and perform tasks. Bash provides a range of features for running commands, managing files, and performing other tasks.
+
+Bash commands perform various tasks within the shell environment. Some common bash commands include ls for listing the contents of a directory, cd for changing the current directory, cp for copying files, mv for moving files, and rm for removing files. Bash also provides a range of advanced commands, such as grep for searching for text within files, sed for performing text transformations, and awk for processing text data.
```
Comment (gshomo) [inline docs/source/using-discovery/bash.rst:10]: maybe "Shell commands perform various tasks with the terminal" ?

Comment (visionjo) [reply docs/source/using-discovery/bash.rst:10]: Are you suggesting in addition or in place of the overview?

**Comment by visionjo** [comment]
A few sections and improvements were made. Please take a look at the 5 commits right above.

File: docs/source/using-discovery/bash.rst [hunk 2]
```diff
@@ -0,0 +1,637 @@
...
+These are just a few examples of Git configuration options. There are many more options available. See `Git User Manual`_ for more information on how to customize Git to your needs.
+
+
+Text Editors
+===============
+There are a few popular text editors that enable modifying text files from the terminal. Here, we provide include emacs, vim, and nano - each are available by default on discovery.
+
+Emacs
+------
+Emacs is a popular text editor that is widely used for programming, writing, and other text-related tasks. Here is a basic tutorial on how to use emacs:
```
Comment (tketchem14) [inline docs/source/using-discovery/bash.rst]: Ideally, we should be specific, not just including "other text-related tasks." I would recommend either expanding the list to be more specific or just removing that example.

File: docs/source/using-discovery/bash.rst [hunk 3]
```diff
@@ -0,0 +1,637 @@
...
+
+Open a terminal and type the following command::
+
+   $ vim
+**Opening a file**
+
+To open an existing file, type the following command::
+
+   vim filename
+**Normal mode**
```
Comment (tketchem14) [inline docs/source/using-discovery/bash.rst:482]: Should this heading be Normal/Insert modes, or something similar?

**Review by tketchem14 — COMMENTED** [review]
Updated the file, left comments where unsure.

File: docs/source/using-discovery/bash.rst [hunk 4]
```diff
@@ -0,0 +1,637 @@
...
+   :alt: Bash (Bourne Again SHell) logo.
+   :align: center
+
+   Bash (Bourne Again SHell)
+
+Overview
+=================
+Bash (Bourne Again SHell) is a popular shell and command-line interface in Unix-like operating systems. A shell is an interface between the user and the underlying operating system, allowing users to interact with the system and perform tasks. Bash provides a range of features for running commands, managing files, and performing other tasks.
+
+Bash commands perform various tasks within the shell environment. Some common bash commands include ``ls`` for listing the contents of a directory, ``cd`` for changing the current directory, ``cp`` for copying files, ``mv`` for moving files, and ``rm`` for removing files. Bash also provides a range of advanced commands, such as ``grep`` for searching for text within files, sed for performing text transformations, and ``awk`` for processing text data. We cover these commands, along with others, in this tutorial.
```
[the remark this answers is not in the document]
Comment (visionjo) [reply docs/source/using-discovery/bash.rst]: I removed definitions and just left them as a list. Overall, the overview was made more straightforward, shorter, and more concise. Thank you.

File: docs/source/using-discovery/bash.rst [hunk 5]
```diff
@@ -0,0 +1,637 @@
...
+
+Terminal
+=================
+
+Mac Users
+---------
+Mac OS comes with a default terminal program. There are various, more advanced terminals available, as well. `iTerm2`_ is one of the more popular choices.
+
+To launch the terminal:
+#. Press Command + Space Bar on your Mac keyboard (alternatively, press F4)
```
[the remark this answers is not in the document]
Comment (visionjo) [reply docs/source/using-discovery/bash.rst:42]: Forgot to add extra line break. Good catch! :)

**Comment by visionjo** [comment]
@keerthanaj98 @joseph-cooke can we merge this in?

**Comment by visionjo** [comment]
@keerthanaj98 @joseph-cooke, can we merge this in? The last time I asked, Tara was on PTO, so my timing there was off. It will be good to get this in asap. Let me know.

**Comment by joseph-cooke** [comment]
@visionjo - I added myself as a reviewer to your PR since you were asking if it was okay to merge. I just wanted to read over it and add a couple of small comments. Thank you!

File: docs/source/using-discovery/bash.rst [hunk 6]
```diff
@@ -0,0 +1,867 @@
...
+   :class: with-border
+   :width: 300
+   :alt: Terminal View.
+   :align: center
+
+   Terminal View
+
+Terminal
+=========
+The terminal - aka the command line interface (CLI) - is a text-based interface for interacting with an operating system. It is a way for users to interact with the system and perform tasks by typing commands and receiving text-based output.
```
Comment (joseph-cooke) [inline docs/source/using-discovery/bash.rst:34]: I would spell out aka - also know as - because someone might not be familiar with the abbreviation.

Comment (joseph-cooke) [reply docs/source/using-discovery/bash.rst:34]: You might what to rephrase the end of this statement because you don't always get a text-based output. cp, mv, and other commands don't give a confirmation after they perform their action. So you might want to say 'typing commands to perform actions' or something like that just because of the commands that don't output text.

File: docs/source/using-discovery/bash.rst [hunk 7]
```diff
@@ -0,0 +1,867 @@
...
+
+Let us explore options and specifics for each operating system; Mac, Linux, and Windows terminals.
+
+MacOS
+------
+Mac OS comes with a default terminal program, but there are more advanced terminals available; `iTerm2`_ is one of the more popular choices.
+
+To launch the terminal:
+
+#. Press Command(⌘) + Space on your Mac keyboard (alternatively, press F4)
```
Comment (joseph-cooke) [inline docs/source/using-discovery/bash.rst:50]: You can rebind the spotlight call from command and space so for some users this might not be the correct combination. You might want to just list the path to terminal 'Macintosh HD/Applications/Utilities/Terminal'.

File: docs/source/using-discovery/bash.rst [hunk 8]
```diff
@@ -0,0 +1,867 @@
...
+#. Type in “Terminal”
+#. When you see Terminal in the Spotlight search list, click it to open the app.
+
+iTerm2 can be installed via the terminal using `Homebrew`_::
```
Comment (joseph-cooke) [inline docs/source/using-discovery/bash.rst:54]: You might want to inform people what Homebrew is just so they know they are installing a package manager to their computer.

File: docs/source/using-discovery/bash.rst [hunk 9]
```diff
@@ -0,0 +1,867 @@
...
+
+Linux
+------
+Linux also comes with a default terminal program, but there are more advanced terminals available; `Terminator`_ is a popular choice.
+
+To download Terminator, open a terminal (``Ctrl+Alt+T`` is the shortcut to do so). Next, execute the following::
+
+   sudo add-apt-repository ppa:gnome-terminator
+   sudo apt-get update
+   sudo apt-get install terminator
```
Comment (joseph-cooke) [inline docs/source/using-discovery/bash.rst:70]: This is for Debian based OSs so you might want to point that out for people who are not using a Debian based OS. For example, there is yum install for Fedora.

File: docs/source/using-discovery/bash.rst [hunk 10]
```diff
@@ -0,0 +1,867 @@
...
+  man ls
+
+The manual page will be displayed in a pager program such as less, which allows you to scroll through the text and search for specific information.
+
+The man command is a valuable tool for learning about new commands and utilities, and it is an essential resource for understanding how to use the terminal effectively. Whether you are a beginner or an advanced user, the man command provides the information you need to get the most out of your tools and utilities.
+
+Basic Commands
+==============
+.. note:: 
+  See `_getting_access` for instructions on having a Discovery user account created.
```
Comment (joseph-cooke) [inline docs/source/using-discovery/bash.rst:93]: The link is not properly showing up in this note.

File: docs/source/using-discovery/bash.rst [hunk 11]
```diff
@@ -0,0 +1,867 @@
...
+
+``cp`` - Copy a file or directory.
+
+.. code-block:: shell
+  :emphasize-lines: 3
+
+   cp file2.txt file3.txt
+   ls
+   file2.txt file3.txt directory1 directory2
+
```
Comment (joseph-cooke) [inline docs/source/using-discovery/bash.rst:172]: You might want to add a comment that it is 'cp <existing file> <copy of existing file>' . It you follow through your example you can see what is being added or removed but if someone just jumps to the copy command they might not see what they are doing with this command.

File: docs/source/using-discovery/bash.rst [hunk 12]
```diff
@@ -0,0 +1,867 @@
...
+``mv`` - Move or rename a file or directory.
+
+.. code-block:: shell
+  :emphasize-lines: 3
+
+  mv file2.txt file4.txt
+  ls
+  file3.txt file4.txt directory1 directory2
+
```
Comment (joseph-cooke) [inline docs/source/using-discovery/bash.rst:184]: Same as above. You might want to just put 'mv <existing file> <renamed existing file>

File: docs/source/using-discovery/bash.rst [hunk 13]
```diff
@@ -0,0 +1,867 @@
...
+  This is the contents of file3.txt
+
+``uniq`` - Remove duplicates from a sorted file or input.
+
+.. code-block:: shell
+  :emphasize-lines: 2
+
+  sort file3.txt | uniq
+  This is the contents of file3.txt
+
```
Comment (joseph-cooke) [inline docs/source/using-discovery/bash.rst:216]: You might want to include a little text to let people know you are piping the output of the sort into the uniq command because some people might not be familiar with |

File: docs/source/using-discovery/bash.rst [hunk 14]
```diff
@@ -0,0 +1,867 @@
...
+  :emphasize-lines: 3
+
+  gunzip file1.txt.gz
+  ls
+  file1.txt
+
+Reference: `gzip(1) manual page`_
+
+``tar`` - Create or extract compressed archive files.
+
```
Comment (joseph-cooke) [inline docs/source/using-discovery/bash.rst:309]: You might want to include a line like:
ls
file1.txt file2.txt

prior to the tar command so it is easier to see that the .tar is new.

Comment (joseph-cooke) [reply docs/source/using-discovery/bash.rst:309]: You might also want to include that you can do entire directories as well since we are suggesting that as a way to get file count down.

File: docs/source/using-discovery/bash.rst [hunk 15]
```diff
@@ -0,0 +1,867 @@
...
+  :emphasize-lines: 2-5
+
+  ls -lh
+  total 8.0K
+  drwxrwxr-x 2 user user 4.0K Feb 14 13:29 directory1
+  -rw-rw-r-- 1 user user   12 Feb 14 13:29 file1.txt
+  -rw-rw-r-- 1 user user   14 Feb 14 13:29 file2.txt
+
+Parameter substitution:
+
```
Comment (joseph-cooke) [inline docs/source/using-discovery/bash.rst:356]: The parameter expansion is hard to follow since name was defined above.

File: docs/source/using-discovery/bash.rst [hunk 16]
```diff
@@ -0,0 +1,867 @@
...
+   diff <(ls /path/to/dir1) <(ls /path/to/dir2)
+
+The next few subsections provide more details on a few advanced bash tools that often come in handy.
+
+rsync
+-----
+The ``rsync`` command is a powerful and versatile file transfer utility commonly used to synchronize files and directories between different locations. It can transfer files over a network connection and run in various modes, including local and remote transfers and backup operations. One of the key benefits of using ``rsync`` is its ability to transfer only the differences between the source and destination files, which can significantly reduce the amount of data transfer time required. Additionally, ``rsync`` supports various advanced features, including the ability to perform incremental backups and preserve symbolic links, making it a popular tool for system administrators and other advanced users.
+
+.. important:: 
+   File transfers must be done using the transfer node on the Discovery, i.e., do not copy to or from the login node accessible via ``xfer.discovery.neu.edu``. See :ref:`transferring data` for more information.
```
Comment (joseph-cooke) [inline docs/source/using-discovery/bash.rst:404]: This statement could be reworded a bit or add in a comma because it seems like you get to the login node by ssh to xfer.

File: docs/source/using-discovery/bash.rst [hunk 17]
```diff
@@ -0,0 +1,867 @@
...
+   find /path/to/dir -name "*.txt"
+
+This will find all files in ``/path/to/dir`` that have a ``.txt`` file extension.
+
+Executing commands on matching files::
+
+   find /path/to/dir -name "*.txt" -exec chmod 644 {} \;
+
+This will find all files in ``/path/to/dir`` that have a ``.txt`` file extension and execute the ``chmod`` command on each file, changing its permissions to ``644``.
+
```
Comment (joseph-cooke) [inline docs/source/using-discovery/bash.rst:480]: You might want to include more information about what chmod is and what does the permission 644 do because some people might not be familiar with this command.

**Comment by joseph-cooke** [comment]
@visionjo - This read really well! I just have a few comments that could addressed to make some items a little clearer. It might also be good to put that you should be running some of these commands on a compute node (like tar on a large directory) rather than the login node too.

File: docs/source/using-discovery/bash.rst [hunk 18]
```diff
@@ -0,0 +1,867 @@
...
+.. code-block:: shell
+  :emphasize-lines: 3
+
+   rm file1.txt
+   ls
+   file2.txt directory1 directory2
+
+.. note:: 
+  To remove a directory, use ``rmdir`` if the folder is empty. Otherwise, recursively delete the directory and all of its contents via ``rm -r <FOLDER_PATH>``.
```
Comment (SerenaCaplins) [inline docs/source/using-discovery/bash.rst:162]: Maybe add a warning that this permanently removes the file or directory?

File: docs/source/using-discovery/bash.rst (hunk 12 above)
Comment (SerenaCaplins) [reply docs/source/using-discovery/bash.rst:184]: Yes I agree with Joseph. I wonder if even putting 'cp' and 'mv' together and saying they work in a similar way as in:

mv input(what to move) output (where/what is it going)

And note that they can use both cp and mv to change a file name (so should proceed with caution....we could probably have a whole best practices page on changing file names)

File: docs/source/using-discovery/bash.rst [hunk 19]
```diff
@@ -0,0 +1,867 @@
...
+.. code-block:: shell
+  :emphasize-lines: 2-4, 7-9
+
+  cat file1.txt
+  This is line 1
+  This is line 2
+  This is line 3
+
+  sed 's/line 1/Line 1/' file1.txt
+  This is Line 1
```
Comment (SerenaCaplins) [inline docs/source/using-discovery/bash.rst:284]: This change was a little subtle. I wonder if you can change the word entirely? sub "line" for "cat" or something more noticible?

File: docs/source/using-discovery/bash.rst [hunk 20]
```diff
@@ -0,0 +1,867 @@
...
+
+Syncing a local directory to a remote server while excluding certain files::
+
+   rsync -avz --exclude='*.log' /local/path user@xfer.discovery.neu.edu:/remote/path
+
+Syncing a remote server to a local directory while preserving symbolic links::
+
+   rsync -avz --links user@xfer.discovery.neu.edu:/remote/path /local/path
+
+find
```
Comment (SerenaCaplins) [inline docs/source/using-discovery/bash.rst:436]: I think we could link to this section in the keeping control of your quota as a way to find those big files that need to be removed or tar-ed.

File: docs/source/using-discovery/bash.rst [hunk 21]
```diff
@@ -0,0 +1,867 @@
...
+  :emphasize-lines: 2-6
+
+  awk '{print NR, $0, length($0)}' awk-example.txt
+  1 John Doe 25 11
+  2 Jane Doe 30 11
+  3 Jim Smith 40 12
+  4 Sara Johnson 35 15
+  5 Michael Brown 29 16
+
+Git configurations tips and tricks
```
Comment (SerenaCaplins) [inline docs/source/using-discovery/bash.rst:633]: Feel like Git should be it's own page
